src/GraphQL/Validation/DocumentValidator.ParseArgumentVisitor.cs

```diff
@@ -17,12 +17,47 @@ private class ParseArgumentVisitor : ASTVisitor<ParseArgumentVisitor.Context>
 
         private ParseArgumentVisitor() { }
 
+        public override ValueTask VisitAsync(ASTNode? node, Context context)
+        {
+            if (node is IHasDirectivesNode hasDirectivesNode)
+            {
+                // if any directives were supplied in the document for the field or fragment spread,
+                // load all defined arguments for directives on the field or fragment spread
+                if (hasDirectivesNode.Directives?.Count > 0)
+                {
+                    try
+                    {
+                        var directives = ExecutionHelper.GetDirectives(hasDirectivesNode, context.Variables, context.Schema);
+                        if (directives != null)
+                        {
+                            (context.DirectiveValues ??= new()).Add(node, directives);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        // todo: report error properly
```

Comment:
the error messages will be replaced in PR #3493 (subsequent to this PR)

Reply:
ok